src/bitset.hpp

```diff
@@ -76,63 +93,144 @@ class Bitset {
+
+  // ** Clade / MultiClade Methods
+  // These methods require bitsets to represent clades. Clades are an expression of a
+  // subset of a taxon set.  The size of the taxon set is equal to the size of the
+  // clade's bitset, with each bit index representing a specific member of that taxon
+  // set.
+  //
+  // There are bitset "types" composed of multiple clades (here, called a
+  // MultiClade). Subsplits are composed of two clades and PCSPs are composed of three
+  // clades.
+  //
+  // Comparator: Clades are sorted with respect to the lexigraphical representation of
+  // their taxon subset. (e.g. If two clades are "010" and "101", then their taxon
+  // subsets are {b} and {a,c}. "b" > "a", therefore "010" > "101".) Note: Sorting by
+  // taxon representation gives the precise opposite ordering to sorting by binary
+  // representation.
+  static int CladeCompare(const Bitset &bitset_a, const Bitset &bitset_b);
+  int CladeCompare(const Bitset &other) const;
+  // For a specified number of clades, return the clade/taxon size.
+  size_t MultiCladeGetCladeSize(const size_t clade_count) const;
+  // For a specified number of clades, return a bitset of the ith clade.
+  Bitset MultiCladeGetClade(const size_t i, const size_t clade_count) const;
+  // For a specified number of clades, return a string of the bitset with each clade
+  // separated by a "|".
+  std::string MultiCladeToString(const size_t clade_count) const;
 
+  // ** Subsplit Methods
   // These methods require the bitset to be a "subsplit bitset" of even length,
-  // consisting of two equal sized "chunks" representing the two sides of the
-  // subsplit.
+  // consisting of two equal sized "clades" representing the two sides of the
+  // subsplit. Clades are stored in a sorted order wrt to their lexigraphic taxon
+  // ordering: the smaller "left" or "sorted" clade stored in the 0th position, and the
+  // larger "right" or "rotated" clade in the 1st position.
   //
+  // Subsplit Constructor: each argument represents one of the clade of the subsplit.
+  // Each clade templated after Bitset constructor.
+  // Build a Subsplit bitset out of a compatible pair of clades.
+  static Bitset Subsplit(const Bitset &clade_0, const Bitset &clade_1);
+  // Builds Clades from string of "1" and "0" characters.
+  static Bitset Subsplit(const std::string clade_0, const std::string clade_1);
+  // Builds Clades of size `n` with only indices in the clade vectors set to true.
+  static Bitset Subsplit(const SizeVector clade_0, const SizeVector clade_1,
+                         const size_t n);
+  // Given two clades, create a subsplit with clades in sorted order by taxon
+  // representation.
+  static Bitset SubsplitOfOrderedClades(const Bitset &clade_0, const Bitset &clade_1);
+  // Comparator: Subsplits are sorting on the following:
+  // (1) The number of taxa in each of their subsplits.
+  // (2) The Bitset ordering of each of their respective unions.
```

Comment:
Yeah these two functions were originally `SubsplitChunk()` and `SubsplitGetChild()`.  Right now, it was written to return the ith clade according to its binary ordering for some reason.  

I think it would be best to have both `SubsplitGetClade` and `SubsplitGetOrderedClade` to return the same output for the same input, just that `SubsplitGetOrderedClade` would assert that the subsplit is indeed ordered, while `SubsplitGetClade` would allow for "rotated" subsplits.